src/retryOnEmpty.test.ts

```diff
@@ -627,6 +628,50 @@ describe('createRetryOnEmptyMiddleware', () => {
       );
     });
   });
+
+  describe('when provider return execution revert error', () => {
+    it('returns the same error to caller', async () => {
+      await withTestSetup(
+        {
+          configureMiddleware: ({ provider, blockTracker }) => {
+            return {
+              middlewareUnderTest: createRetryOnEmptyMiddleware({
+                provider,
+                blockTracker,
+              }),
+            };
+          },
+        },
+        async ({ engine, provider }) => {
+          const request: JsonRpcRequest<string[]> = {
+            id: 123,
+            jsonrpc: '2.0',
+            method: 'eth_call',
+            params: buildMockParamsWithoutBlockParamAt(2, '100'),
+          };
+          stubProviderRequests(provider, [
+            buildStubForBlockNumberRequest(),
+            {
+              request,
+              response: () => {
+                throw {
+                  code: errorCodes.rpc.invalidInput,
+                  message: 'execution reverted',
+                } as any;
```

Comment:
`rpc-errors` can build the errors too. To avoid the type assertion, what do you think about:

``` suggestion
                throw rpcErrors.invalidInput('execution reverted');
```

(note: you'll have to update the `import` at the top)